src/main/groovy/com/orctom/gradle/archetype/util/FileUtils.groovy

```diff
@@ -69,44 +73,104 @@ class FileUtils {
 
     logBindings(binding)
 
+    // list of written files
+    // currently used only for cleaning up when there is a conflict and ConflictResolutionStrategy is FAIL
+    List<File> filesWritten = new ArrayList<>();
+
     templates.each {
       source ->
-      try {
+
+        // apply variable substitution to path
         File target = new File(targetDir, resolvePaths(getRelativePath(sourceDir, source)))
         String path = engine.createTemplate(target.path).make(binding)
         target = new File(path)
-        boolean isFile = source.isFile()
-        ensureParentDirs(target, isFile)
 
-        if (isFile) {
-          if (isNotTemplate(source.path, nonTemplates)) {
-            Files.copy(source.toPath(), target.toPath())
+        if (source.isFile()) {
+
+          // ensure ancestor dirs exist
+          target.mkdirs();
+
+          if (isNonTemplate(source, nonTemplates)) {
+            writeNonTemplate(target, source, strategy, filesWritten)
           } else {
-            try {
-              target << resolve(source.text, binding)
-            } catch (Exception ex) {
-              log.error("Failed to resolve variables in: '{}]", source.path)
-              log.error(ex.getMessage())
-              Files.copy(source.toPath(), target.toPath())
-            }
+            writeTemplate(target, source, strategy, binding, filesWritten)
           }
         }
-      } catch (Exception e) {
-        e.printStackTrace()
-      }
     }
 
     log.info('Done')
   }
 
-  static def logBindings(Map map) {
+  // handle templates
+  private static void writeTemplate(File target, File source, ConflictResolutionStrategy strategy,
+                                    Map binding, List<File> filesWritten ) {
+    try {
+
+      if (target.exists()) {
+        switch (strategy) {
+
+          case OVERWRITE:
+            log.info("Overwriting file '{}'.", target.absolutePath)
+            target.delete()
+            target << resolve(source.text, binding)
+            Files.copy(source.toPath(), target.toPath(), StandardCopyOption.REPLACE_EXISTING)
+            break
+
+          case FAIL:
+            log.error("File already exists '{}'.", target.absolutePath)
+            log.info("Stopping the generation, deleting generated files.")
+            // remove generated files
+            filesWritten.each { file -> file.delete() }
+            System.exit(1)
+            break
+        }
+      } else {
+        target << resolve(source.text, binding)
+      }
+
+      filesWritten.add(target)
+
+    } catch (Exception ex) {
+      log.error("Failed to resolve variables in: '{}]", source.path)
+      log.error(ex.getMessage())
+      Files.copy(source.toPath(), target.toPath())
+    }
+  }
+
+  // handle non-templates
+  private static void writeNonTemplate(File target, File source, ConflictResolutionStrategy strategy, List<File> filesWritten ) {
+
+    if (target.exists()) {
+      switch (strategy) {
+
+        case OVERWRITE:
```

Comment:
I prefer not to support this strategy to _Keep It Simple and Stupid_. 
But yes it could be reasonable, let me know how would you use this plugin, and how these strategies matters.